commit 7e979e4922e39ae01a75ff416329541ffaeb3911 (refs/changes/12/2812/2)
Author: Sam Gwydir <sam@samgwydir.com>
Date:   2017-12-04T11:23:59-08:00 (1 year, 10 months ago)
    
    MANATEE-374 Want configuration changes for improved observability, heartbeating and logging

diff --git a/etc/9.6/postgresql.manta.coal.conf b/etc/9.6/postgresql.manta.coal.conf
index dc6a3da..4791322 100644
--- a/etc/9.6/postgresql.manta.coal.conf
+++ b/etc/9.6/postgresql.manta.coal.conf
@@ -429,13 +429,13 @@ log_min_messages = error                # values in order of decreasing detail:
 #debug_print_rewritten = off
 #debug_print_plan = off
 #debug_pretty_print = on
-#log_checkpoints = off
-#log_connections = off
-#log_disconnections = off
+log_checkpoints = on
+log_connections = on
+log_disconnections = on
 #log_duration = off
 #log_error_verbosity = default          # terse, default, or verbose messages
 #log_hostname = off
-log_line_prefix = '%m'                  # special values:
+log_line_prefix = '%t [%p]: [%l-1] user=%u,db=%d,app=%a,client=%h ' # special values:
                                         #   %a = application name
                                         #   %u = user name
                                         #   %d = database name
@@ -456,10 +456,10 @@ log_line_prefix = '%m'                  # special values:
                                         #        processes
                                         #   %% = '%'
                                         # e.g. '<%u%%%d> '
-#log_lock_waits = off                   # log lock waits >= deadlock_timeout
+log_lock_waits = on                     # log lock waits >= deadlock_timeout
 #log_statement = 'none'                 # none, ddl, mod, all
 #log_replication_commands = off
-#log_temp_files = -1                    # log temporary files equal or larger
+log_temp_files = 0                      # log temporary files equal or larger
                                         # than the specified size in kilobytes;
                                         # -1 disables, 0 logs all temp files
 log_timezone = 'UTC'
@@ -480,8 +480,8 @@ log_timezone = 'UTC'
 
 #track_activities = on
 #track_counts = on
-#track_io_timing = off
-#track_functions = none                 # none, pl, all
+track_io_timing = on
+track_functions = all                   # none, pl, all
 #track_activity_query_size = 1024       # (change requires restart)
 #stats_temp_directory = 'pg_stat_tmp'
 
@@ -500,13 +500,10 @@ log_timezone = 'UTC'
 
 #autovacuum = on                        # Enable autovacuum subprocess?  'on'
                                         # requires track_counts to also be on.
-#log_autovacuum_min_duration = -1       # -1 disables, 0 logs all actions and
+log_autovacuum_min_duration = 0         # -1 disables, 0 logs all actions and
                                         # their durations, > 0 logs only
                                         # actions running at least this number
                                         # of milliseconds.
-log_autovacuum_min_duration = 1000
-          # set to 1 second to log any vacuum operations that take longer than
-          # a second. Per Pg High Perf page 163
 #autovacuum_max_workers = 3             # max number of autovacuum subprocesses
                                         # (change requires restart)
 #autovacuum_naptime = 1min              # time between autovacuum runs
@@ -652,3 +649,4 @@ default_text_search_config = 'pg_catalog.english'
 #------------------------------------------------------------------------------
 
 # Add settings for extensions here
+pg_stat_statements.track = 'all'
diff --git a/etc/9.6/postgresql.manta.lab.conf b/etc/9.6/postgresql.manta.lab.conf
index d745fe8..1b2f464 100644
--- a/etc/9.6/postgresql.manta.lab.conf
+++ b/etc/9.6/postgresql.manta.lab.conf
@@ -429,13 +429,13 @@ log_min_messages = error                # values in order of decreasing detail:
 #debug_print_rewritten = off
 #debug_print_plan = off
 #debug_pretty_print = on
-#log_checkpoints = off
-#log_connections = off
-#log_disconnections = off
+log_checkpoints = on
+log_connections = on
+log_disconnections = on
 #log_duration = off
 #log_error_verbosity = default          # terse, default, or verbose messages
 #log_hostname = off
-log_line_prefix = '%m'                  # special values:
+log_line_prefix = '%t [%p]: [%l-1] user=%u,db=%d,app=%a,client=%h ' # special values:
                                         #   %a = application name
                                         #   %u = user name
                                         #   %d = database name
@@ -456,10 +456,10 @@ log_line_prefix = '%m'                  # special values:
                                         #        processes
                                         #   %% = '%'
                                         # e.g. '<%u%%%d> '
-#log_lock_waits = off                   # log lock waits >= deadlock_timeout
+log_lock_waits = on                     # log lock waits >= deadlock_timeout
 #log_statement = 'none'                 # none, ddl, mod, all
 #log_replication_commands = off
-#log_temp_files = -1                    # log temporary files equal or larger
+log_temp_files = 0                      # log temporary files equal or larger
                                         # than the specified size in kilobytes;
                                         # -1 disables, 0 logs all temp files
 log_timezone = 'UTC'
@@ -480,8 +480,8 @@ log_timezone = 'UTC'
 
 #track_activities = on
 #track_counts = on
-#track_io_timing = off
-#track_functions = none                 # none, pl, all
+track_io_timing = on
+track_functions = all                   # none, pl, all
 #track_activity_query_size = 1024       # (change requires restart)
 #stats_temp_directory = 'pg_stat_tmp'
 
@@ -500,13 +500,10 @@ log_timezone = 'UTC'
 
 #autovacuum = on                        # Enable autovacuum subprocess?  'on'
                                         # requires track_counts to also be on.
-#log_autovacuum_min_duration = -1       # -1 disables, 0 logs all actions and
+log_autovacuum_min_duration = 0         # -1 disables, 0 logs all actions and
                                         # their durations, > 0 logs only
                                         # actions running at least this number
                                         # of milliseconds.
-log_autovacuum_min_duration = 1000
-          # set to 1 second to log any vacuum operations that take longer than
-          # a second. Per Pg High Perf page 163
 #autovacuum_max_workers = 3             # max number of autovacuum subprocesses
                                         # (change requires restart)
 #autovacuum_naptime = 1min              # time between autovacuum runs
@@ -652,3 +649,4 @@ default_text_search_config = 'pg_catalog.english'
 #------------------------------------------------------------------------------
 
 # Add settings for extensions here
+pg_stat_statements.track = 'all'
diff --git a/etc/9.6/postgresql.manta.production.conf b/etc/9.6/postgresql.manta.production.conf
index 72f5d84..7455cee 100644
--- a/etc/9.6/postgresql.manta.production.conf
+++ b/etc/9.6/postgresql.manta.production.conf
@@ -429,13 +429,13 @@ log_min_messages = error                # values in order of decreasing detail:
 #debug_print_rewritten = off
 #debug_print_plan = off
 #debug_pretty_print = on
-#log_checkpoints = off
-#log_connections = off
-#log_disconnections = off
+log_checkpoints = on
+log_connections = on
+log_disconnections = on
 #log_duration = off
 #log_error_verbosity = default          # terse, default, or verbose messages
 #log_hostname = off
-log_line_prefix = '%m'                  # special values:
+log_line_prefix = '%t [%p]: [%l-1] user=%u,db=%d,app=%a,client=%h ' # special values:
                                         #   %a = application name
                                         #   %u = user name
                                         #   %d = database name
@@ -456,10 +456,10 @@ log_line_prefix = '%m'                  # special values:
                                         #        processes
                                         #   %% = '%'
                                         # e.g. '<%u%%%d> '
-#log_lock_waits = off                   # log lock waits >= deadlock_timeout
+log_lock_waits = on                     # log lock waits >= deadlock_timeout
 #log_statement = 'none'                 # none, ddl, mod, all
 #log_replication_commands = off
-#log_temp_files = -1                    # log temporary files equal or larger
+log_temp_files = 0                      # log temporary files equal or larger
                                         # than the specified size in kilobytes;
                                         # -1 disables, 0 logs all temp files
 log_timezone = 'UTC'
@@ -480,8 +480,8 @@ log_timezone = 'UTC'
 
 #track_activities = on
 #track_counts = on
-#track_io_timing = off
-#track_functions = none                 # none, pl, all
+track_io_timing = on
+track_functions = all                   # none, pl, all
 #track_activity_query_size = 1024       # (change requires restart)
 #stats_temp_directory = 'pg_stat_tmp'
 
@@ -500,13 +500,10 @@ log_timezone = 'UTC'
 
 #autovacuum = on                        # Enable autovacuum subprocess?  'on'
                                         # requires track_counts to also be on.
-#log_autovacuum_min_duration = -1       # -1 disables, 0 logs all actions and
+log_autovacuum_min_duration = 0         # -1 disables, 0 logs all actions and
                                         # their durations, > 0 logs only
                                         # actions running at least this number
                                         # of milliseconds.
-log_autovacuum_min_duration = 1000
-          # set to 1 second to log any vacuum operations that take longer than
-          # a second. Per Pg High Perf page 163
 #autovacuum_max_workers = 3             # max number of autovacuum subprocesses
                                         # (change requires restart)
 #autovacuum_naptime = 1min              # time between autovacuum runs
@@ -653,3 +650,4 @@ default_text_search_config = 'pg_catalog.english'
 #------------------------------------------------------------------------------
 
 # Add settings for extensions here
+pg_stat_statements.track = 'all'
